scenario/uploaddata.go

```diff
@@ -120,11 +114,17 @@ func (settings UploadDataSettings) Execute(
 		uploadCtx = ctx
 		defer cancel()
 	}
+	file, err := os.Open(settings.Filename)
+	if err != nil {
+		actionState.AddErrors(errors.Wrapf(err, "failed to open file <%s>", settings.Filename))
+		return
+	}
 	tempFile, err := tempFileClient.UploadFromFile(uploadCtx, file)
 	if err != nil {
 		actionState.AddErrors(errors.Wrap(err, "failed to upload temp content from file"))
 		return
 	}
+	_ = file.Close()
```

Comment:
defer close after open, otherwise it's not closed on e.g. error from UploadFromFile method